airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -147,7 +146,7 @@ public void start() {
           }
         });
 
-    final WorkerFactory factory = WorkerFactory.newInstance(WorkflowClient.newInstance(temporalService));
+    final WorkerFactory factory = TemporalClient.productionWorkerFactory(configs);
```

Comment:
In the same vein as comments above, reworked this abstraction to let `TemporalClient` decide if it should talk to Temporal Cloud, or `airbyte-temporal`, based on `configs.temporalCloudEnabled`